### PR TITLE
fix: use docker stop not kill

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

use docker stop not kill

## Why is it important?

To kill the container directly is a little overwhelming, the default signal for `docker kill` is KILL

### CODE REVIEW

File: vars/filebeat.groovy [hunk 1]
```diff
@@ -81,7 +81,7 @@ def stop(Map args = [:]){
   // due to the harcoded creation mask, see https://github.com/elastic/beats/issues/20584
   sh(label: 'Stop filebeat', script: """
     docker exec -t ${config.id} chmod -R ugo+rw /output || echo "Exit code \$?"
-    docker kill ${config.id} || echo "Exit code \$?"
+    docker stop --time 30 ${config.id} || echo "Exit code \$?"
```
Comment (mdelapenya) [inline vars/filebeat.groovy:84]: This will mean we are not removing the container because the worker is ephemeral, right?

**Comment by mdelapenya** [comment]
For reference, we saw this error on Jenkins trying to store a file that was not closed yet

>ava.io.IOException: Truncated TAR archive
	at org.apache.commons.compress.archivers.tar.TarArchiveInputStream.read(TarArchiveInputStream.java:643)
	at java.io.InputStream.read(InputStream.java:101)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:2314)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:2270)
	at org.apache.commons.io.IOUtils.copyLarge(IOUtils.java:2291)
	at org.apache.commons.io.IOUtils.copy(IOUtils.java:2246)
	at hudson.util.IOUtils.copy(IOUtils.java:43)
	at hudson.FilePath.readFromTar(FilePath.java:2641)
Also:   	Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to JNLP4-connect connection from beats-ci-immutable-ubuntu-1804-1610377061675462028.c.elastic-ci-prod.internal/10.224.0.160:44246
			at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1800)
			at hudson.remoting.UserRequest$ExceptionResponse.retrieve(UserRequest.java:357)
			at hudson.remoting.Channel$2.adapt(Channel.java:1037)
			at hudson.remoting.Channel$2.adapt(Channel.java:1033)
			at hudson.remoting.FutureAdapter.get(FutureAdapter.java:64)
			at hudson.FilePath.copyRecursiveTo(FilePath.java:2432)
			at jenkins.model.StandardArtifactManager.archive(StandardArtifactManager.java:70)
			at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:257)
			at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:116)
			at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:71)
			at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
			at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
			at java.util.concurrent.FutureTask.run(FutureTask.java:266)
			at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
			at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.io.IOException: This archive contains unclosed entries.
		at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.finish(TarArchiveOutputStream.java:289)
		at org.apache.commons.compress.archivers.tar.TarArchiveOutputStream.close(TarArchiveOutputStream.java:307)
		at hudson.util.io.TarArchiver.close(TarArchiver.java:126)
		at hudson.FilePath.writeToTar(FilePath.java:2611)
		at hudson.FilePath.access$4200(FilePath.java:212)
		at hudson.FilePath$CopyRecursiveRemoteToLocal.invoke(FilePath.java:2574)
		at hudson.FilePath$CopyRecursiveRemoteToLocal.invoke(FilePath.java:2561)
		at hudson.FilePath$FileCallableWrapper.call(FilePath.java:3112)
		at hudson.remoting.UserRequest.perform(UserRequest.java:212)
		at hudson.remoting.UserRequest.perform(UserRequest.java:54)
		at hudson.remoting.Request$2.run(Request.java:369)
		at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:72)
		at java.util.concurrent.FutureTask.run(FutureTask.java:266)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
		at hudson.remoting.Engine$1.lambda$newThread$0(Engine.java:93)
	Caused: java.util.concurrent.ExecutionException
		at hudson.remoting.Channel$2.adapt(Channel.java:1039)
		at hudson.remoting.Channel$2.adapt(Channel.java:1033)
		at hudson.remoting.FutureAdapter.get(FutureAdapter.java:64)
		at hudson.FilePath.copyRecursiveTo(FilePath.java:2432)
Caused: java.io.IOException: Failed to extract /var/lib/jenkins/workspace/e2e-tests_e2e-testing-mbp_PR-577/transfer of 1 files
	at hudson.FilePath.readFromTar(FilePath.java:2651)
	at hudson.FilePath.copyRecursiveTo(FilePath.java:2429)
	at jenkins.model.StandardArtifactManager.archive(StandardArtifactManager.java:70)
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:257)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:116)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:71)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

File: vars/filebeat.groovy (hunk 1 above)
Comment (kuisathaverat) [reply vars/filebeat.groovy:84]: it is not removed on both cases

Comment (mdelapenya) [reply vars/filebeat.groovy:84]: Good point, it's not `docker container remove`, thanks!

**Review by v1v — APPROVED** [review]

**Review by mdelapenya — APPROVED** [review]

File: vars/filebeat.groovy (hunk 1 above)
Comment (mdelapenya) [inline vars/filebeat.groovy:84]: As a follow up, the timeout could be configurable, with 30 as fallback, just in case the consumer needs another time 🤷

Comment (kuisathaverat) [reply vars/filebeat.groovy]: done

**Review by cachedout — APPROVED** [review]

**Pull request merged**